### PR TITLE
Cabal test int test

### PR DESCRIPTION
**Pull request by kquick**
Run intTests via a `cabal test` operation.  Mirrors/replaces `runtest.sh` functionality.

### CODE REVIEW

**Comment by kquick** [comment]
Current status: waiting on merge of PR #668 which will fix test issues to allow final validation of the changes here. Merge of PR #668 is delayed due to a bike test regression.

**Comment by robdockins** [comment]
I think this should be unblocked now with the merge of #840

File: .gitignore
Comment (RyanGlScott) [inline .gitignore:23]: How did you determine what files from `intTests` to put in `.gitignore`?

Comment (kquick) [reply .gitignore:23]: Chatted w/Ryan, this is in reference to #1067.  @LisannaAtGalois , we could perhaps optimistically invoke a make clean after each target, ignoring failures (e.g. no clean target), but we should probably also make sure the gitignore us up-to-date with the latest test outputs.

**Comment by LisannaAtGalois** [comment]
This is now behaving functionally equivalently to the runtests.sh implementation - fixing the timeout to match what's currently used has all the results lining up now. The last thing this PR needs is to make the test details print on 100% success, and to remove runtests.sh.

Follow-up work to smooth out how the integration tests are themselves specified to make it more cohesive with it being a cabal test target would be awesome:
- Move saw's Main.hs into a library, invoke main function directly from the tests (which should still include all the command-line argument parsing). 
- Output results in a standard test result format, and then actually do something a bit smarter than just grepping that file afterwards. Regression-comparing the results against master and gating PRs on observed test regression is a really nice way to "automate" the upkeep of PR gating criteria compared to manually-curated lists of disabled / "known failing" cases and the `continue-on-error` sledgehammer.

File: .gitignore
Comment (LisannaAtGalois) [reply .gitignore:23]: I'll take a look and manually update the .gitignore (if needed). Maybe this could be automatically linted in CI by running all the tests (or their makefiles?) and checking to see if the git tree is dirty.

Invoking make clean at least on test success makes sense to me. Probably better to have failing tests not invoke that to preserve the output for post-mortem analysis.

File: intTests/IntegrationTest.hs
Comment (kquick) [inline intTests/IntegrationTest.hs]: We should probably keep this comment.

**Comment by LisannaAtGalois** [comment]
Pending successful CI after merging in master, this should be ready to go in now.

This does intersect a bit with https://github.com/GaloisInc/saw-script/pull/1111, which I will update after this is merged.

**Review by atomb — APPROVED** [review]
This looks great. Thanks!

**Pull request merged**
